lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_send_app_size_metrics.rb

```diff
@@ -0,0 +1,165 @@
+            variant_descriptors.each do |desc|
+              variant_metadata = { device: desc['device'], 'OS Version': desc['os-version'] }
+              metrics_helper.add_metric(name: 'Download Size', value: variant['sizeCompressedApp'], meta: variant_metadata)
+              metrics_helper.add_metric(name: 'Install Size', value: variant['sizeUncompressedApp'], meta: variant_metadata)
+            end
+          end
+        end
+
+        # Send the payload
+        metrics_helper.send_metrics(
+          to: api_url,
+          api_token: api_token,
+          use_gzip: params[:use_gzip_content_encoding]
+        )
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        'Send iOS app size metrics to our metrics server'
+      end
+
+      def self.details
+        <<~DETAILS
+          Send iOS app size metrics to our metrics server.
+
+          In order to get Xcode generate the `app-thinning.plist` file (during `gym` and the export of the `.xcarchive`), you need to:
+            (1) Use either `ad-hoc`, `enterprise` or `development` export method (in particular, won't work with `app-store`),
+            (2) Provide `thinning: '<thin-for-all-variants>'` as part of your `export_options` of `gym` (or in your `options.plist` file if you use raw `xcodebuild`)
+          See https://help.apple.com/xcode/mac/11.0/index.html#/devde46df08a
+
+          For builds exported with the `app-store` method, `xcodebuild` won't generate an `app-thinning.plist` file; so you will only be able to get
+          the Universal `.ipa` file size as a metric, but won't get the per-device, broken-down install and download sizes for each thinned variant.
+
+          See https://github.com/Automattic/apps-metrics for the API contract expected by the Metrics server you are expected to send those metrics to.
+
+          Tip: If you provide a `file://` URL for the `api_url`, the action will write the payload on disk at the specified path instead of sending
+          the data to a endpoint over network. This can be useful e.g. to inspect the payload and debug it, or to store the metrics data as CI artefacts.
+        DETAILS
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :api_url,
+            env_name: 'FL_IOS_SEND_APP_SIZE_METRICS_API_URL',
+            description: 'The endpoint API URL to publish metrics to. (Note: you can also point to a `file://` URL to write the payload to a file instead)',
+            type: String,
+            optional: false
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :api_token,
+            env_name: 'FL_IOS_SEND_APP_SIZE_METRICS_API_TOKEN',
+            description: 'The bearer token to call the API. Required, unless `api_url` is a `file://` URL',
```

Comment:
If we go with my suggestion to make this parameter required, we could update the description to

```diff
-            description: 'The bearer token to call the API. Required, unless `api_url` is a `file://` URL',
+            description: 'The bearer token to call the API. Ignored if `api_url` is a `file://` URL', 
```

Reply:
If we go with parameter being required, that means the call site will still **have** to provide a (dummy) non-`nil` value for the `api_token`… even if it gets completely unused… seems wrong and confusing to me, API-design-wise?

Reply:
> seems wrong and confusing to me, API-design-wise?

Granted. I considered it acceptable under the assumption that using `file://` was not an "official" use case, which I now appreciate was an incorrect assumption, as per https://github.com/wordpress-mobile/release-toolkit/pull/364/files#r880888176